galleries/examples/ticks/date_index_formatter.py

```diff
@@ -25,17 +25,17 @@
 # low, close, volume, adj_close from the mpl-data/sample_data directory. The
 # record array stores the date as an np.datetime64 with a day unit ('D') in
 # the date column (``r.date``).
-r = cbook.get_sample_data('goog.npz')['price_data'].view(np.recarray)
+r = cbook.get_sample_data('goog.npz')['price_data']
 r = r[:9]  # get the first 9 days
 
 fig, (ax1, ax2) = plt.subplots(nrows=2, figsize=(6, 6), layout='constrained')
 fig.get_layout_engine().set(hspace=0.15)
 
 # First we'll do it the default way, with gaps on weekends
-ax1.plot(r.date, r.adj_close, 'o-')
+ax1.plot(r["date"], r["adj_close"], 'o-')
 
 # Highlight gaps in daily data
-gaps = np.flatnonzero(np.diff(r.date) > np.timedelta64(1, 'D'))
+gaps = np.flatnonzero(np.diff(r["date"]) > np.timedelta64(1, 'D'))
 for gap in r[['date', 'adj_close']][np.stack((gaps, gaps + 1)).T]:
     ax1.plot(gap.date, gap.adj_close, 'w--', lw=2)
```

Comment:
This one is a copy of `r` that hasn't been changed.
```suggestion
    ax1.plot(gap['date'], gap['adj_close'], 'w--', lw=2)
```